src/main/kotlin/io/github/inductiveautomation/kindling/idb/tagconfig/model/TagConfig.kt

```diff
@@ -0,0 +1,153 @@
+package io.github.inductiveautomation.kindling.idb.tagconfig.model
+
+import kotlinx.serialization.ExperimentalSerializationApi
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.json.JsonArray
+import kotlinx.serialization.json.JsonElement
+import kotlinx.serialization.json.JsonNull
+import kotlinx.serialization.json.JsonObject
+import kotlinx.serialization.json.JsonPrimitive
+import kotlinx.serialization.json.JsonTransformingSerializer
+import kotlinx.serialization.json.jsonObject
+
+@Serializable
+data class TagConfig(
+    // Basic Properties:
+    val name: String? = null,
+    val tagGroup: JsonElement? = null, // String
+    val enabled: JsonElement? = null, // String
+    // Value Properties:
+    val tagType: String? = null, // Unlisted
+    val typeId: String? = null, // Unlisted
+    val valueSource: String? = null,
+    val dataType: JsonElement? = null, // String
+    val value: JsonElement? = null, // JsonPrimitive
+    val opcServer: JsonElement? = null, // OPC, String
+    val opcItemPath: JsonElement? = null, // OPC // JsonElement
+    val sourceTagPath: JsonElement? = null, // Derived, Reference, String
+    val executionMode: JsonElement? = null, // String
+    val executionRate: JsonElement? = null, // Int
+    val expression: JsonElement? = null, // Expression, String
+    @SerialName("deriveExpressionGetter")
```

Comment:
These members can't be the underlying types for some reason?

Reply:
Some of these were marked `JsonElement` because we didn't care about them. Others were marked as such because they sort-of had to be.
For example, opcItemPath can be just a string, or it can be a JSON Object with keys `bindType` and `binding`.

We _could_ make a class that represents that. But meh.

Reply:
Fair enough :) Maybe just add a comment about why they're not the primitives to prevent someone from getting overzealous in the future